### ISSUE TITLE
Questions about Updating Memory Module

### ISSUE THREAD

**Issue by crazyn2** [cvlab-yonsei/MNAD#57]
I've read your paper which is an excellent work. But when I was reading your code, I have questions about the procedures of updating memory module. 
From my perspective, the m_items in Train.py is the memory module, whose grad is False. So the backward operations of the separateness loss and compactness_loss do not have any impact on memory. However, the paper discusses how these two loss functions can influence memory. The reason for including these two loss functions in the overall loss of Train.py is not explicitly stated.

**Comment by crazyn2** [comment]
Terrible codes in this repository

**Comment by myccver** [comment]
我的理解是，虽然memory不可以被梯度更新，但是代码里通过前向传播不断更新memory参数。注意separateness loss and compactness_loss的梯度可以更新encoder，也就是说可以不断影响encoder编码的特征，特征又会对memory更新产生影响。所以，这两个loss也会对memory产生影响

**Comment by alchemi5t** [comment]
https://arxiv.org/abs/2101.12382

We also have a repo with experiments on the memory module. Feel free to check it out.
Repo linked in paper.

**Comment by crazyn2** [comment]
> 我的理解是，虽然memory不可以被梯度更新，但是代码里通过前向传播不断更新memory参数。注意的梯度可以更新encoder，也就是说可以不断影响encoder编码的特征，特征又会对memory更新产生影响。所以，这两个loss也会对memory产生影响

你可以看下代码里面的detach, 然后把这两个损失函数删掉，看看效果，说不定你会发现更好的效果

**Comment by myccver** [comment]
> > 我的理解是，虽然memory不可以被梯度更新，但是代码里通过前向传播不断更新memory参数。注意的梯度可以更新encoder，也就是说可以不断影响encoder编码的特征，特征又会对memory更新产生影响。所以，这两个loss也会对memory产生影响
> 
> 你可以看下代码里面的detach, 然后把这两个损失函数删掉，看看效果，说不定你会发现更好的效果

我是直接去掉了整个memory，使用提供的默认参数在重建任务上取得了比论文中完整模型更好一点的效果。作者的消融有些问题，我怀疑故意压低baseline效果

![Image](https://github.com/user-attachments/assets/3cf4be7b-4606-4c06-86cf-830efacd5fbe)

![Image](https://github.com/user-attachments/assets/3646d446-236d-4845-9a6b-844b108b2cfe)

**Comment by crazyn2** [comment]
是的，我之前就是因为发现了这点，你才60轮，如果跑更多轮次，应该可以再提升些

**Comment by yinyexingchen** [comment]
> 是的，我之前就是因为发现了这点，你才60轮，如果跑更多轮次，应该可以再提升些

你好，请问我在复现的时候，运行Train.py过后，CPU和GPU都在运转，但是控制台一直没有输出，log里也什么都没有，看起来像是没有在训练呢

**Comment by crazyn2** [comment]
> > 是的，我之前就是因为发现了这点，你才60轮，如果跑更多轮次，应该可以再提升些
> 
> 你好，请问我在复现的时候，运行Train.py过后，CPU和GPU都在运转，但是控制台一直没有输出，log里也什么都没有，看起来像是没有在训练呢

正常的，只要GPU和CPU在跑就是在训练，好像默认就是没有，输出的代码需要你加

**Comment by yinyexingchen** [comment]
> > > 是的，我之前就是因为发现了这点，你才60轮，如果跑更多轮次，应该可以再提升些
> > 
> > 
> > 你好，请问我在复现的时候，运行Train.py过后，CPU和GPU都在运转，但是控制台一直没有输出，log里也什么都没有，看起来像是没有在训练呢
> 
> 正常的，只要GPU和CPU在跑就是在训练，好像默认就是没有，输出的代码需要你加

您好，就是我看到那个训练代码里有每轮训练打印epoch，但是我的运行代码过后等待很久也没有打印的，log里也没有epoch的信息，似乎一直卡在加载数据这个地方。还有想请教一下您，我使用更高的pyotrch版本是否可以呢，1.1.0的版本在windows下似乎没办法配置呀

**Comment by crazyn2** [comment]
> > > > 是的，我之前就是因为发现了这点，你才60轮，如果跑更多轮次，应该可以再提升些
> > > 
> > > 
> > > 你好，请问我在复现的时候，运行Train.py过后，CPU和GPU都在运转，但是控制台一直没有输出，log里也什么都没有，看起来像是没有在训练呢
> > 
> > 正常的，只要GPU和CPU在跑就是在训练，好像默认就是没有，输出的代码需要你加
> 
> 您好，就是我看到那个训练代码里有每轮训练打印epoch，但是我的运行代码过后等待很久也没有打印的，log里也没有epoch的信息，似乎一直卡在加载数据这个地方。还有想请教一下您，我使用更高的pyotrch版本是否可以呢，1.1.0的版本在windows下似乎没办法配置呀

wimdow系统就这样，卡在数据加载很久，可以用ubuntu
